lib/CLI/Types.hs

```diff
@@ -167,35 +209,168 @@ data FieldInfo = FieldInfo
   { fiName :: FieldName
   , fiContents :: FieldContents
   }
-  deriving stock Show
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
 
 data Direction = Asc | Desc
-  deriving stock Show
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
 
 data Sort
   = SortByEntryName
   | SortByEntryDate
   | SortByFieldContents FieldName
   | SortByFieldDate FieldName
-  deriving stock Show
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
 
 data FilterOp = OpGT | OpGTE | OpLT | OpLTE | OpEQ
-  deriving stock Show
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
 
 data FilterDate
   = FDYear Year
   | FDMonth Month
   | FDDay Day
   | FDTime UTCTime
-  deriving stock Show
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
 
 data Filter
   = FilterByDate FilterOp FilterDate
   | FilterByName Text
   | FilterByField FieldName FilterField
-  deriving stock Show
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
 
 data FilterField
   = FilterFieldByDate FilterOp FilterDate
   | FilterFieldByContents Text
-  deriving stock Show
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
+----------------------------------------------------------------------------
+-- Instances
+----------------------------------------------------------------------------
+
+instance FromHttpApiData Filter where
+  parseUrlPiece = toServantParser do
+    choice
+      [ do
+          void "name~"
+          name <- takeRest
+          guard $ not $ T.null name
+          return $ FilterByName name
+      , do
+          void "date"
+          op <- choice
+            [ ">=" $> OpGTE
+            , "<=" $> OpLTE
+            , ">"  $> OpGT
+            , "<"  $> OpLT
+            , "="  $> OpEQ
+            ]
+          FilterByDate op <$> parseFilterDate
+      ]
```

Comment:
Is the handling of the third constructor, `FilterByField`, is omitted here on purpose?

Reply:
Hmm. I think I forgot to handle it after rebasing. But I think it could be solved when we implement #89

Reply:
Agree, we just have not to forget about this. Maybe put some todo comment?